### PR TITLE
[Chef] Add Air Quality Sensor device type

### PR DESCRIPTION
**Pull request by erwinpan1**
Create Chef Air Quality Sensor device type

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: examples/chef/common/chef-concentration-measurement.cpp [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+#include <app/clusters/concentration-measurement-server/concentration-measurement-server.h>
+using namespace chip::app::Clusters::ConcentrationMeasurement;
+#endif
+
+#ifdef EMBER_AF_PLUGIN_CARBON_DIOXIDE_CONCENTRATION_MEASUREMENT_SERVER
+static Instance gCarbonDioxideConcentrationMeasurementInstance =
+    CreateNumericMeasurementAndLevelIndicationConcentrationCluster<true, true, true, true>(
+        EndpointId(1), CarbonDioxideConcentrationMeasurement::Id, MeasurementMediumEnum::kAir, MeasurementUnitEnum::kPpm);
+
+void emberAfCarbonDioxideConcentrationMeasurementClusterInitCallback(EndpointId endpoint)
```
Comment (bzbarsky-apple) [inline examples/chef/common/chef-concentration-measurement.cpp:46]: Is there guaranteed to be only one endpoint involved, as this code seems to assume?  Might be good to VerifyOrDie() that assumption.

File: examples/chef/devices/rootnode_airqualitysensor_e63187f6c9.matter [hunk 2]
```diff
@@ -0,0 +1,2165 @@
...
+    ram      attribute identifyType default = 0x00;
+    callback attribute generatedCommandList;
+    callback attribute acceptedCommandList;
+    callback attribute eventList;
+    callback attribute attributeList;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 4;
+  }
+
+  server cluster Groups {
```
Comment (bzbarsky-apple) [inline examples/chef/devices/rootnode_airqualitysensor_e63187f6c9.matter:1905]: What is groups doing here?  It's not part of the Air Quality Sesor device type.

File: examples/chef/devices/rootnode_airqualitysensor_e63187f6c9.matter [hunk 3]
```diff
@@ -0,0 +1,2165 @@
...
+    callback attribute levelValue default = 1;
+    callback attribute generatedCommandList;
+    callback attribute acceptedCommandList;
+    callback attribute eventList;
+    callback attribute attributeList;
+    ram      attribute featureMap default = 63;
+    ram      attribute clusterRevision default = 3;
+  }
+
+  server cluster Pm10ConcentrationMeasurement {
```
Comment (bzbarsky-apple) [inline examples/chef/devices/rootnode_airqualitysensor_e63187f6c9.matter:2104]: I assume "PM2.5 Concentration Measurement" is disabled here on purpose?

File: examples/chef/common/chef-concentration-measurement.cpp (hunk 1 above)
Comment (erwinpan1) [reply examples/chef/common/chef-concentration-measurement.cpp:46]: Yes. Added `VerifyOrDie()`

File: examples/chef/devices/rootnode_airqualitysensor_e63187f6c9.matter (hunk 3 above)
Comment (erwinpan1) [reply examples/chef/devices/rootnode_airqualitysensor_e63187f6c9.matter:2075]: No. The [PM2.5](https://github.com/project-chip/connectedhomeip/pull/29436/files#diff-8b1b09c1c7162c3750a2c9ea83273564c3b54bb87f1a803410d19e5e256c08a2R2015-R2033) which is generated in front of PM1 which follows the ZAP file's order

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**